src/pal/src/misc/time.cpp

```diff
@@ -394,3 +394,40 @@ GetTickCount64()
     return retval;
 }
 
+/*++
+Function:
+  PAL_nanosleep
+
+Sleeps for the time specified in requestedTime.  If the operation is interrupted by a signal,
+remainingTime will contain the amount of time left from the original request.  Returns 0 on
+successful completion of the operation.
+--*/
+PALAPI
+INT
+PAL_nanosleep(
+    IN TIMESPEC *requestedTime,
+    OUT TIMESPEC *remainingTime
+    )
+{
+    struct timespec req;
+    struct timespec rem;
+    int result;
+
+    if(requestedTime == NULL)
+    {
+        return -1;
+    }
+
+    req.tv_sec = requestedTime->Seconds;
+    req.tv_nsec = requestedTime->Nanoseconds;
+
+    result = nanosleep(&req, &rem);
```

Comment:
I also don't have a strong opinion here. My feeling was that it is not worth adding a new type just to be able to pass the argument that can specify long times when I don't see a usage for a non-interruptible sleep longer than 1s.